packages/react-component-library/src/components/Checkbox/partials/StyledCheckmark.tsx

```diff
@@ -64,16 +64,18 @@ export const StyledCheckmark = styled.div<CheckmarkProps>`
     }
   }
 
-  ${StyledCheckbox} input:disabled ~ & {
+  ${StyledCheckbox} input:disabled ~ && {
```

Comment:
The meaning of `&` and `&&` depend on the context: https://styled-components.com/docs/basics#pseudoelements-pseudoselectors-and-nesting

It needed to be `&&` here because of the conditional logic – otherwise the container and no-container styles would get mixed up if both are on the same page.